storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdates.java

```diff
@@ -266,6 +269,36 @@ public void applyToStore(final Store store) {
                 removeBlockRootFromSlotIndex(store.rootsBySlotLookup, slot, root);
               });
         });
+
+    // Update forkchoice
+    applyForkChoiceUpdates(store);
+  }
+
+  private void applyForkChoiceUpdates(final Store store) {
+    // Process new blocks and states
+    final List<SignedBeaconBlock> sortedBlocks = new ArrayList<>(hotBlocks.values());
+    sortedBlocks.sort(Comparator.comparing(SignedBeaconBlock::getSlot));
+    for (SignedBeaconBlock newBlock : sortedBlocks) {
+      final BeaconState newState = hotStates.get(newBlock.getRoot());
+      store.forkChoiceState.onBlock(new SignedBlockAndState(newBlock, newState));
```

Comment:
nit: No need to wrap and unwrap the block and state here:
```suggestion
      store.forkChoiceState.onBlock(newBlock.getMessage(), newState);
```